### PR TITLE
strider 1.6.3 (new formula)

### PR DESCRIPTION
**Pull request by sc0ttyd**
Added formula for Strider CD, the open source
continuous integration and deployment server
http://stridercd.com

### CODE REVIEW

File: Library/Formula/strider.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
+class UniversalNpm < Requirement
```
Comment (mikemcquaid) [inline Library/Formula/strider.rb]: Should just be `NpmRequirement` as it's not universal.

File: Library/Formula/strider.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  fatal true
```
Comment (mikemcquaid) [inline Library/Formula/strider.rb]: `default_formula "node"` in here.

File: Library/Formula/strider.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  sha1 "6f0fe2b306e2bce395d39b9ce2dc6249bdcf616d"
+
+  depends_on UniversalNpm
+  depends_on "mongodb"
+
+  head "https://github.com/Strider-CD/strider.git", :branch => "develop"
+
+  def install
+    system "npm", "install"
+    prefix.install Dir["bin", "client", "lib", "public", "node_modules", "bower_components"], "main.js", "package.json"
```
Comment (mikemcquaid) [inline Library/Formula/strider.rb]: Don't need to use `Dir` here

File: Library/Formula/strider.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    if File.file?(etc/"striderrc")
+      ohai "Existing config file found at #{HOMEBREW_PREFIX}/etc/striderrc"
+    else
+      (etc/"striderrc").write conf_file
```
Comment (mikemcquaid) [inline Library/Formula/strider.rb]: Use `(buildpath/"striderrc").write` and then `etc.install` which will stop the existing file being overwritten.

File: Library/Formula/strider.rb [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      ohai "The strider config file has been installed to #{HOMEBREW_PREFIX}/etc/striderrc. Please modify this after the installation."
+    end
+  end
+
+  def caveats
+    "If this is a clean install, you will need to add an admin user. To do so, execute 'strider addUser' and follow the prompts."
```
Comment (mikemcquaid) [inline Library/Formula/strider.rb]: This needs wrapped so it's printed nicely with an 80 character terminal. Also, unless this is Homebrew-specific it should probably be removed.

File: Library/Formula/strider.rb [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        "smtp_from" : "Strider <noreply@stridercd.com>",
+        "enablePty"      : false,
+        "extpath"        : "node_modules",
+        "session_secret" : "8L8BudMkqBUqrz",
+        "github_app_id"  : "",
+        "github_secret"  : "",
+        "cors"           : false
+      }
+    EOS
+  end
```
Comment (mikemcquaid) [inline Library/Formula/strider.rb]: Is this provided by upstream somewhere? Would rather avoid hardcoding this if possible.

File: Library/Formula/strider.rb [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        <string>#{var}/log/strider.log</string>
+        <key>StandardOutPath</key>
+        <string>#{var}/log/strider.log</string>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    system "strider", "-v"
```
Comment (mikemcquaid) [inline Library/Formula/strider.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by jacknagel** [comment]
Closing due to no response.

**Pull request closed without merging**